egs/dampvsep/ConvTasNet/train.py

```diff
@@ -0,0 +1,168 @@
+import os
+import argparse
+from pathlib import Path
+import json
+
+import torch
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.data import DataLoader
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping
+
+from asteroid.models import ConvTasNet
+from asteroid.data import DAMPVSEPDataset
+from asteroid.engine.optimizers import make_optimizer
+from asteroid.engine.system import System
+from asteroid.losses import SingleSrcNegSTOI
+
+
+# Keys which are not in the conf.yml file can be added here.
+# In the hierarchical dictionary created when parsing, the key `key` can be
+# found at dic['main_args'][key]
+# By default train.py will use all available GPUs. The `id` option in run.sh
+# will limit the number of available GPUs for train.py .
+parser = argparse.ArgumentParser()
+parser.add_argument("--exp_dir", default="exp/tmp", help="Full path to save best validation model")
+
+
+def main(conf):
+    # Define dataloader using ORIGINAL mixture.
+    dataset_kwargs = {
+        "root_path": Path(conf["data"]["root_path"]),
+        "task": conf["data"]["task"],
+        "sample_rate": conf["data"]["sample_rate"],
+        "num_workers": conf["training"]["num_workers"],
+        "mixture": conf["data"]["mixture"],
+    }
+
+    train_set = DAMPVSEPDataset(
+        split=f"train_{conf['data']['train_set']}",
+        random_segments=True,
+        segment=conf["data"]["segment"],
+        samples_per_track=conf["data"]["samples_per_track"],
+        **dataset_kwargs,
+    )
+
+    val_set = DAMPVSEPDataset(split="valid", **dataset_kwargs)
+
+    train_loader = DataLoader(
+        train_set,
+        shuffle=True,
+        batch_size=conf["training"]["batch_size"],
+        num_workers=conf["training"]["num_workers"],
+        drop_last=True,
+    )
+
+    val_loader = DataLoader(
+        val_set, shuffle=False, batch_size=1, num_workers=conf["training"]["num_workers"]
+    )
+
+    conf["masknet"].update({"n_src": conf["data"]["n_src"]})
+
+    model = ConvTasNet(**conf["filterbank"], **conf["masknet"])
+    optimizer = make_optimizer(model.parameters(), **conf["optim"])
+
+    # Define scheduler
+    scheduler = None
+    if conf["training"]["half_lr"]:
+        scheduler = ReduceLROnPlateau(optimizer=optimizer, factor=0.5, patience=5)
+    # Just after instantiating, save the args. Easy loading in the future.
+    exp_dir = conf["main_args"]["exp_dir"]
+    os.makedirs(exp_dir, exist_ok=True)
+    conf_path = os.path.join(exp_dir, "conf.yml")
+    with open(conf_path, "w") as outfile:
+        yaml.safe_dump(conf, outfile)
+
+    # Define Loss function.
+    # Combine_Loss is not complete. Needs improvement
+    # loss_func = Combine_Loss(alpha=conf['training']['loss_alpha'],
+    #                          sample_rate=conf['data']['sample_rate'])
+    loss_func = torch.nn.L1Loss()
```

Comment:
Do you want to fix this before merging?

Reply:
No need to wait. When using remix mixture, L1Loss works ok.
The problems are when using the original mixture.
This is why in run.sh I have 
> "mixture=remix

What do you prefer?

Reply:
I let you decide about that. 
This seems fine.

Reply:
Thank you @mpariente 
I think it is ready for merging. If I manage to fix the loss function, I can create another PR with a modification of the recipe.
Perhaps, one recipe that selects the loss function depending on the mixture (remix, original) or I can add a parameter to select the loss function.

Reply:
I'll try to review today, thanks